tartufo/scanner.py

```diff
@@ -351,18 +357,23 @@ def rule_matches(rule: Rule, string: str, line: str, path: str) -> bool:
 
         :param rule: Rule to perform match
         :param string: string to match against rule pattern
+        :param line: Source line containing string of interest
         :param path: path to match against rule path_pattern
         :return: True if string and path matched, False otherwise.
         """
         match = False
-        if rule.re_match_type == "match":
+        if rule.re_match_scope == Scope.Word:
+            scope = string
+        else:
+            scope = line
+        if rule.re_match_type == MatchType.Match:
             if rule.pattern:
-                match = rule.pattern.match(string) is not None
+                match = rule.pattern.match(scope) is not None
             if rule.path_pattern:
                 match = match and rule.path_pattern.match(path) is not None
-        elif rule.re_match_type == "search":
+        elif rule.re_match_type == MatchType.Search:
             if rule.pattern:
-                match = rule.pattern.search(line) is not None
+                match = rule.pattern.search(scope) is not None
             if rule.path_pattern:
                 match = match and rule.path_pattern.search(path) is not None
```

Comment:
Similar to above, we should probably error out for invalid values now that we're formalizing the ability to configure this functionality.